src/encoder.rs

```diff
@@ -1166,17 +1166,19 @@ pub fn encode_block_b(fi: &FrameInvariants, fs: &mut FrameState,
         }
     }
 
-    if skip {
-        cw.bc.reset_skip_context(bo, bsize, xdec, ydec);
-    }
-
     // these rules follow TX_MODE_LARGEST
     let tx_size = match bsize {
         BlockSize::BLOCK_4X4 => TxSize::TX_4X4,
         BlockSize::BLOCK_8X8 => TxSize::TX_8X8,
         BlockSize::BLOCK_16X16 => TxSize::TX_16X16,
         _ => TxSize::TX_32X32
     };
+    cw.bc.set_tx_size(bo, tx_size);
+    // Were we not hardcoded to TX_MODE_LARGEST, block tx size would be written here
```

Comment:
This is very good point.

If TX_MODE_SELECT is signaled, for intra block, tx_size is required even if the block is skipped.
So, tx_size must be signaled and also written to tx_size mode book keeping.